### ISSUE TITLE
Integer isn't numeric in bitwise or (|)'

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4780]
Migrated from [rt.perl.org#8174](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=8174) (status was 'resolved')

Searchable as RT8174$

**Comment by p5pRT** [comment]
### From dk@plab.ku.dk

#### Created by dk@plab.ku.dk



<details><summary>Perl Info</summary>

```


This perlbug was built using Perl 5.00503 - Sun Mar  5 13:39:27 SAST 2000
It is being executed now by  Perl 5.006001 - Wed Jan  2 14:08:06 CET 2002.

Site configuration information for perl 5.006001:

Configured by dk at Wed Jan  2 14:08:06 CET 2002.

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=freebsd, osvers=4.3-stable, archname=i386-freebsd
    uname='freebsd plab.ku.dk 4.3-stable freebsd 4.3-stable #1: fri nov 23 11:33:13 cet 2001 root@plab.ku.dk:usrobjusrsrcsyspuma i386 '
    config_args='-sde -Dprefix=/usr/local -Darchlib=/usr/local/lib/perl5/5.6.1/i386-freebsd -Dprivlib=/usr/local/lib/perl5/5.6.1 -Dman3dir=/usr/local/lib/perl5/5.6.1/man/man3 -Dsitearch=/usr/local/lib/perl5/site_perl/5.6.1/i386-freebsd -Dsitelib=/usr/local/lib/perl5/site_perl/5.6.1 -Ui_gdbm'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -I/usr/local/include',
    optimize='-O -pipe ',
    cppflags='-fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='2.95.3 [FreeBSD] 20010315 (release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags ='-Wl,-E  -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib
    libs=-lgdbm -ldb -lm -lc -lcrypt -liconv -lutil -ldmalloc
    perllibs=-lm -lc -lcrypt -liconv -lutil -ldmalloc
    libc=, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-DPIC -fpic', lddlflags='-shared  -L/usr/local/lib'

Locally applied patches:
    


@INC for perl 5.006001:
    /usr/home/dk/src/Prima
    /usr/home/dk/src/IPC
    /home/dk/src/perl-5.6.1/lib
    /usr/local/lib/perl5/5.6.1
    /usr/local/lib/perl5/site_perl/5.6.1/i386-freebsd
    /usr/local/lib/perl5/site_perl/5.6.1
    /usr/local/lib/perl5/site_perl/5.005/i386-freebsd
    /usr/local/lib/perl5/site_perl/5.005
    /usr/local/lib/perl5/site_perl
    .


Environment for perl 5.006001:
    HOME=/home/dk
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/X11R6/bin:/usr/local/site/bin:/home/dk/bin:/plab/usr/home/dk/bin:/plab/usr/local/site/bin:/home/dk/bin/kuim
    PERL5LIB=/usr/home/dk/src/Prima:/usr/home/dk/src/IPC
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/zsh




```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Dmitry Karasik

Thanks for the report\.

> 
> This is a bug report for perl from dk@&#8203;plab\.ku\.dk\,
> generated with the help of perlbug 1\.26 running under perl 5\.006001\.
> 
> Hello 
> 
> I found the response of perl\-5\.7\.1 on the following code interesting&#8203;:
> 
> use constant INT => 1;
> my @&#8203;x = \( INT|INT => 1 \);
> 
> and perl curses 'Argument "INT" isn't numeric in bitwise or \(|\)'
> 
> although if the last string is changed to
> 
> my @&#8203;x = \( INT|INT \, 1 \);
> 
> everything is fine\. Well it's not a severe bug but it might
> be something\, especially because => and \, have equal precedences
> \( at least after perlop\.pod \)\.

The have equal precedence\, but not the same semantic\. :\-\(

=> quotes any immediately preceding bareword\, so in the first example the 
second INT is a quoted bareword "INT" not the constant sub &INT evaluating to 1\.  
\( \(INT|INT\) => 1 \) should work\.

This is documented in the last paragraph of the BUGS section of constant\.pm

Robin

\-\- 
Robin Barker                        | Email&#8203;: Robin\.Barker@&#8203;npl\.co\.uk
CMSC\, Building 10\,                  | Phone&#8203;: \+44 \(0\) 20 8943 7090
National Physical Laboratory\,       | Fax&#8203;:   \+44 \(0\) 20 8977 7091
Teddington\, Middlesex\, UK\. TW11 OLW | WWW&#8203;:   http&#8203;://www\.npl\.co\.uk

**Comment by p5pRT** [comment]
### From @dk

Please unsubscribe me

Sincerely\,
&nbsp;  Dmitry Karasik

\-\-\-\-\-\-\-\- Original Message \-\-\-\-\-\-\-\-
Received&#8203;: \(from root@&#8203;localhost\)by plab\.ku\.dk \(8\.11\.6/8\.9\.3\) id
g2ID8Tv54415for dk@&#8203;plab\.ku\.dk\.AVP; Mon\, 18 Mar 2002 14&#8203;:08&#8203;:29 \+0100
\(CET\)\(envelope\-from perlbug@&#8203;rfi\.net\)
Received&#8203;: from onion\.perl\.org \(onion\.valueclick\.com \[209\.85\.157\.220\]\)by
plab\.ku\.dk \(8\.11\.6/8\.9\.3\) with SMTP id g2ID8R854407for \<dk@&#8203;plab\.ku\.dk>;
Mon\, 18 Mar 2002 14&#8203;:08&#8203;:28 \+0100 \(CET\)\(envelope\-from perlbug@&#8203;rfi\.net\)
Received&#8203;: \(qmail 45363 invoked by uid 1007\); 18 Mar 2002 13&#8203;:08&#8203;:23 \-0000
Date&#8203;: 18 Mar 2002 13&#8203;:08&#8203;:23 \-0000
Message\-ID&#8203;: \<20020318130823\.45362\.qmail@&#8203;onion\.perl\.org>
From&#8203;: perlbug@&#8203;onion\.perl\.org
Subject&#8203;: Perlbug \- reminder of bug\(20020104\.002\) status
X\-Errors\-To&#8203;: perlbug@&#8203;rfi\.net
To&#8203;: dk@&#8203;plab\.ku\.dk
X\-Perlbug&#8203;: Perlbug\(tron\) v2\.95
X\-SpamBouncer&#8203;: 1\.4 \(10/07/01\)
X\-SBClass&#8203;: OK
Status&#8203;: 
X\-Mozilla\-Status&#8203;: 8001
X\-Mozilla\-Status2&#8203;: 00000000
X\-UIDL&#8203;: 3c95e91800000001


&nbsp;        This is a Perlbug status\(open\) reminder for an outstanding 
&nbsp;        bug\, a report for which is appended at the base of this email\.

&nbsp;        If the the status of this bug is in any way incorrect\, please
&nbsp;        inform an administrator of the Perlbug system\.

&nbsp;        Further data relating to this bug\(20020104\.002\) may be found at&#8203;:

&nbsp;            http&#8203;://bugs\.perl\.org/perlbug\.cgi?req=bidmid&bidmid=20020104\.002

&nbsp;        The group\(\) of administrators responsible for this bug is&#8203;:

&nbsp;            http&#8203;://bugs\.perl\.org/perlbug\.cgi?req=group\_id&group\_id=

&nbsp;        For email help send an email to&#8203;:

&nbsp;            To&#8203;: bugdb@&#8203;perl\.org
&nbsp;            Subject&#8203;: \-H

&nbsp;        Bug report \(current status\) follows&#8203;:    
&nbsp;        

Bug&#8203;:      20020104\.002     
Subject&#8203;:  Integer isn't numeric in bitwise or \(|\)'
From&#8203;:     Dmitry Karasik \<dk@&#8203;plab\.ku\.dk>
Created&#8203;:  2002\-01\-04 06&#8203;:14&#8203;:29  Modified&#8203;: 2002\-01\-04 06&#8203;:14&#8203;:29

Status&#8203;:   open             Severity&#8203;: 
OS names&#8203;: freebsd
Group&#8203;:    
Version&#8203;:  5\.061\, 5\.006001  Fixed in&#8203;:  

Message ids&#8203;: 1      
Patch ids&#8203;:   0     
Test ids&#8203;:    0    
Note ids&#8203;:    0    
Change ids&#8203;:  0    
Parent ids&#8203;:  0    
Child ids&#8203;:   0    
Address ids&#8203;: 0

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
